lib/plugins/aws/package/lib/mergeIamTemplates.js

```diff
@@ -66,7 +73,14 @@ module.exports = {
     iamRoleLambdaExecutionTemplate.Properties.Path = this.provider.naming.getRolePath();
     iamRoleLambdaExecutionTemplate.Properties.RoleName = this.provider.naming.getRoleName();
 
-    if (this.serverless.service.provider.rolePermissionsBoundary) {
+    // set permission boundary
+    if (iamRole.permissionBoundary) {
+      iamRoleLambdaExecutionTemplate.Properties.PermissionsBoundary = iamRole.permissionBoundary;
+    } else if (this.serverless.service.provider.rolePermissionsBoundary) {
+      this.serverless._logDeprecation(
+        'PROVIDER_IAM_SETTINGS',
+        'Starting with version 3.0.0, "provider.rolePermissionsBoundary" property will be replaced with "provider.iam.role.permissionBoundary"'
+      );
       iamRoleLambdaExecutionTemplate.Properties.PermissionsBoundary = this.serverless.service.provider.rolePermissionsBoundary;
```

Comment:
If users sets both `provider.iam.role.permissionBoundary` and `provider.rolePermissionsBoundary`, we should favor first (which we do now), but we still should show information that deprecated property was configured (It's important to do it, as introducing not supported properties with next version will crash the deployment)